tests/conftest.py

```diff
@@ -120,8 +120,14 @@ def pytest_collection_modifyitems(items):
         http://pytest.org/latest/plugins.html
     """
     for item in items:
-        if 'qtbot' in getattr(item, 'fixturenames', ()):
+        if 'qapp' in getattr(item, 'fixturenames', ()):
             item.add_marker('gui')
+            if sys.platform == 'linux' and not os.environ.get('DISPLAY', ''):
+                if 'CI' in os.environ:
+                    raise Exception("No display available on CI!")
```

Comment:
This is a bit of a hack (and will be displayed as an `INTERNALERROR>`), but I think there's no other good way to mark the tests as failed from here - and since this never should happen I'm okay with it.